test/unit/service/authServiceTests.ts

```diff
@@ -1,49 +1,88 @@
-    mock.onPost('http://localhost:8080/api/login', loginData).reply(401);
+    it('should handle invalid login credentials', async () => {
+      var mock = new MockAdapter(axios);
+      const loginData = { email: 'email@email.com', password: 'invalidPassword' };
+      mock.onPost(`${process.env.API_URL}api/login`, loginData).reply(401);
 
-    try {
-      await login(loginData);
-    } catch (error) {
+      try {
+        await authService.login(loginData);
+      } catch (error) {
         console.log(error.message)
-      expect(error.message).to.equal('Invalid credentials - 401');
-    }
-  });
+        expect(error.message).to.equal('Invalid credentials - 401');
+      }
+    });
 
-  it('should handle server error', async () => {
-    var mock = new MockAdapter(axios);
-    const loginData = { email: 'email@email.com', password: 'invalidPassword' };
-    mock.onPost('http://localhost:8080/api/login', loginData).reply(500, 'Internal Server Error');
+    it('should handle server error', async () => {
+      var mock = new MockAdapter(axios);
+      const loginData = { email: 'email@email.com', password: 'invalidPassword' };
+      mock.onPost(`${process.env.API_URL}api/login`, loginData).reply(500, 'Internal Server Error');
 
-    try {
-      await login(loginData);
-    } catch (error) {
-      expect(error.message).to.equal('Internal server error - 500');
-    }
+      try {
+        await authService.login(loginData);
+      } catch (error) {
+        expect(error.message).to.equal('Internal server error - 500');
+      }
+    });
   });
+
+  describe('whoami', () => {
+    it('should identify user successfully', async () => {
+      const mock = new MockAdapter(axios);
+      const token = 'token';
+      const responseData: User = { id: 1, email: 'test@test.com', role: 'Employee' };
+      mock.onGet(`${process.env.API_URL}api/whoami`, { headers: { Authorization: `Bearer ${token}` } }).reply(200, responseData);
+        
+      const result = await authService.whoami(token);
+      expect(result).to.deep.equal(responseData);
+    });
+    
+    it('should handle error when user is not logged in', async () => {
+      const mock = new MockAdapter(axios);
+      const token = 'token';
+      mock.onGet(`${process.env.API_URL}api/whoami`, { headers: { Authorization: `Bearer ${token}` } }).reply(401);
+        
+      try {
+        await authService.whoami(token);
+      } catch (error) {
+        expect(error.message).to.equal('User is not logged in');
+      }
+    });
+    
+    it('should handle error when server error', async () => {
+      const mock = new MockAdapter(axios);
+      const token = 'token';
+      mock.onGet(`${process.env.API_URL}api/whoami`, { headers: { Authorization: `Bearer ${token}` } }).reply(500);
+        
+      try {
+        await authService.whoami(token);
+      } catch (error) {
+        expect(error.message).to.equal('Failed to fetch user');
```

Comment:
You should be able to use expect throws to check for exceptions and then you don't need the try/catch blocks